Source/WebCore/Modules/filesystemaccess/WorkerFileSystemStorageConnection.cpp

```diff
@@ -379,4 +379,31 @@ void WorkerFileSystemStorageConnection::move(FileSystemHandleIdentifier identifi
     });
 }
 
+std::optional<uint64_t> WorkerFileSystemStorageConnection::requestNewCapacityForSyncAccessHandle(FileSystemHandleIdentifier identifier, FileSystemSyncAccessHandleIdentifier accessHandleIdentifier, uint64_t newCapacity)
+{
+    if (!m_scope)
+        return std::nullopt;
+
+    if (!m_mainThreadConnection)
+        return std::nullopt;
+
+    BinarySemaphore semaphore;
+    std::optional<uint64_t> grantedCapacity;
+    callOnMainThread([mainThreadConnection = m_mainThreadConnection, identifier, accessHandleIdentifier, newCapacity, &grantedCapacity, &semaphore]() {
```

Comment:
It would be nice if we would not need to go through main thread since it might get stuck and this IPC call might be time sensitive.
We could have the binary semaphore.signal be called on a work queue instead.
This could be left to a follow-up patch maybe.

Reply:
I don't quite understand this comment. Do you mean web process should send out NetworkStorageManager::RequestNewCapacityForSyncAccessHandle message on a WorkQueue instead of on main thread? How would this work?